include/settings/settings.h

```diff
@@ -248,15 +269,57 @@ void settings_dst_register(struct settings_store *cs);
 /**
  * Parses a key to an array of elements and locate corresponding module handler.
  *
- * @param name Key in string format
- * @param name_argc Parsed number of elements.
- * @param name_argv Parsed array of elements.
+ * @param[in] name in string format
+ * @param[out] next remaining of name after matched handler
  *
  * @return settings_handler node on success, NULL on failure.
  */
-struct settings_handler *settings_parse_and_lookup(char *name, int *name_argc,
-						   char *name_argv[]);
+struct settings_handler *settings_parse_and_lookup(const char *name,
+						   const char **next);
+
+
+/*
+ * API for const name processing
+ */
+
+/**
+ * Compares the start of name with a key
+ *
+ * @param[in] name in string format
+ * @param[in] key comparison string
+ * @param[out] next pointer to remaining of name (excluding separator)
+ *
+ * @return 0: no match
+ *         1: match, next can be used to check if match is full
+ */
+int settings_name_cmp(const char *name, const char *key, const char **next);
 
+/**
+ * Splits up name in part before and after first separator ('/')
+ *
+ * @param[in] name in string format
+ * @param[out] argv part of name before first separator ('/')
+ * @param[out] next pointer to remaining of name (excluding separator)
+ *
+ * @return 0: there is no separator in name, argv contains name
+ *         1: there is a separator in name
+ */
+int settings_name_split(const char *name, char *argv, const char **next);
+
+/**
+ * Splits up name in count of items separated by separator ('/') and a list
+ * of the separated items
+ *
+ * @param[in] name in string format
```

Comment:
[in,out] as the function modifies the given string.

Reply:
Is this the way we are support to document parameter in Zephyr though? It doesn't looks like kernel.h doesn't make use of in,out at all.

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/e4892abe31df6730357db50f671935991606bf2c/subsys/bluetooth/controller/util/memq.c#L86

And a lot in externals directory.
We have a few [in/out] too - but as far as I know, it would not work. I will check it out and possibly generate another issue.

Reply:
But wait a moment - direction type after a name?? This should not work either.

Reply:
It sounds a bit ambiguous to me and one still would need to describe how the input and output will be used, and then there is struct parameters where each field can be either input or output or both, anyway Id stick with kernel.h style so if we decide to change we consistently change for everything.

Reply:
function removed

Reply:
Funny.